### ISSUE TITLE
[FR] rename LR scheduler base class _LRScheduler to be non-private

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#61232]
## 🚀 Feature
Rename `_LRScheduler` to something like `LRScheduler` without underscore. 

## Motivation

Allows for writing typed code much better. We have `torch.optim.Optimizer` but no public base class for learning rate schedulers.

**Comment by SsnL** [comment]
This is such a small issue with a huge user experience improvement. Can we bump this?

**Comment by JJBT** [comment]
Any progress?

**Comment by mikaylagawarecki** [comment]
Hello @JJBT, thanks for bumping! We would accept a patch that fixes this.
